common/Unit.hpp

```diff
@@ -356,6 +362,7 @@ class UnitBase
     UnitType _type;
 
     std::mutex _lock; //< Used to protect cleanup functions.
+    std::mutex _lockSocketPoll; //< Used to sync _socketPoll
```

Comment:
Surprised to see a new mutex; fewer mutexes are really a good thing - we could use _lock to protect both the setup and cleanup surely ?

Reply:
Not really a surprise, see my PR offer (flavor, taste) https://github.com/CollaboraOnline/online/pull/10244